vizro-core/src/vizro/models/_components/graph.py

```diff
@@ -121,22 +122,14 @@ def _filter_interaction(
     @staticmethod
     def _optimise_fig_layout_for_dashboard(fig):
         """Post layout updates to visually enhance charts used inside dashboard."""
-        # Reduce `margin_t` if no title is provided and `margin_t` is not explicitly set.
-        if fig.layout.margin.t is None and fig.layout.title.text is None:
-            fig.update_layout(margin_t=24, title_pad_l=0, title_pad_r=0, margin_l=24)
-
-        # Reduce `title_pad_t` if no subtitle is provided and `title_pad_t` is not explicitly set.
-        if fig.layout.title.pad.t is None and fig.layout.title.text and "<br>" not in fig.layout.title.text:
-            fig.update_layout(title_pad_t=7)
-
-        if fig.layout.title.pad.l is None:
-            fig.update_layout(title_pad_l=0)
-
-        if fig.layout.title.pad.r is None:
-            fig.update_layout(title_pad_r=0)
-
-        if fig.layout.margin.l is None:
-            fig.update_layout(margin_l=24)
+        if fig.layout.title.text:
```

Comment:
@huong-li-nguyen the idea here is that I've made the `dashboard_overrides` optimised for a chart with no title so `margin_t=24` there. Hence the logic here now just tweaks things when we have a title.

In future when we have `header` I guess we will just remove this whole `_optimise` method?

Reply:
If you think this is a bad way to do it (e.g. if we should keep `margin_t=64` and not override it in `dashboard_overrides`) then please do say 🙂

Reply:
No, that works for me! That they have a title is probably the rarer case, so this way around also works (and will be called less)!